subsys/mgmt/mcumgr/lib/mgmt/src/mgmt.c

```diff
@@ -23,6 +23,7 @@ mgmt_streamer_alloc_rsp(struct mgmt_streamer *streamer, const void *req)
 void
 mgmt_streamer_trim_front(struct mgmt_streamer *streamer, void *buf, size_t len)
 {
+	streamer->reader->message_size -= len;
```

Comment:
What if `streamer->reader->message_size < len`? This seems possible according to:
```
/**
 * @brief Uses the specified streamer to trim data from the front of a buffer.
 *
 * If the amount to trim exceeds the size of the buffer, the buffer is
 * truncated to a length of 0.
 *
 * @param streamer              The streamer providing the callback.
 * @param buf                   The buffer to trim.
 * @param len                   The number of bytes to remove.
 */
void mgmt_streamer_trim_front(struct mgmt_streamer *streamer, void *buf,
                              size_t len);
```

Second thing is: I am not sure that this is the right place to modify `streamer->reader`. Why this member and not `streamer->writer`? Documentation of this function does not say that it can be used only when reading and when reader is initialized.

Reply:
I will move the stream modification to the `smp_process_request_packet`. The thing is broken in many places and requires some significant rework; the function does not even tell which stream it works on because it actually  uses buf which is some unidentified net_buf.

Reply:
Done, changed smp_process_request_packet  instead.